### ISSUE TITLE
PlayerNotificationManager: customizeable icons

### ISSUE THREAD

**Issue by nomisRev** [google/ExoPlayer#6266]
### Use case description
I'd like to customize the icons currently statically fixed in `PlayerNotificationManager`, more specifically I'd customize `ACTION_REWIND` and `ACTION_FAST_FORWARD` but I believe that all icons should be customizable.

I'd be happy to contribute and create a PR with a proposed solution from below or any other proposed solution. I also wanted to write some documentation for [PlayerNotificationManager] for [https://exoplayer.dev/](https://exoplayer.dev/) since I struggled reading through issues and code but couldn't find where that was possible.

### Proposed solution
- Accept a POJO or `Map<String, NotificationCompat.Action>` in the constructor to specify `resId`

##### Alternative solutions
- Allow `CustomActionReceiver` to override the default actions. [related code](https://github.com/google/ExoPlayer/blob/51acd8150c25a2064c3e0c49e04540bd4e2db6f9/library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerNotificationManager.java#L979).
- Elaborate `CustomActionReceiver` to include the default actions, this way the user can take over control over the actions completely if necessary while maintaining the same defaults.

### Alternatives considered
- I looked at the source code and there is currently no way to work around this because the actions are `private final` fields and the `createPlaybackActions` function is static. Additionally, I think this should stay in the constructor. 

Thanks for this great lib!

**Comment by ojw28** [comment]
I think you can just add `exo_notification_rewind` and `exo_notification_fastforward` drawables to your app resources, and these will override the ones in the ExoPlayer library. This approach works for me in our demo app.

Could you give that a try? If that's sufficient then I think you already have what you need, although you're right that we should document this properly.

**Comment by nomisRev** [comment]
@ojw28 I confirmed that this works. I completely forgot that I could overwrite the drawable in my app and that it'd get overwritten during merging. Thanks!

I'd be happy to add some documentation on this if you could tell me where I can contribute to the documentation.

I've noticed some other things with `PlayerNotificationManger` tho. I am still investigating what the issues are tho.

 - the chronometer flickers on rewind but doesn't on fast-forward.
 - notification is not dismissable when paused, which made it linger forever. (need to pinpoint the issue, might be related to my WakeLock, WifiLock or foreground service logic).

**Comment by ojw28** [comment]
It's definitely dismiss-able when paused for me. If you're setting it as the notification for a foreground service then that might change things because displaying a notification is a requirement for a foreground service (you'd probably need to have the service exit the foreground for it to become dismiss-able again).

We fixed some notification flicker issues in 2.10.0, but they only affected fairly old devices (KitKat), so that's probably not what you're seeing.

For documentation, I think any additions would go into https://exoplayer.dev/ui-components.html. Notification manager could be a section similar to the `PlayerView` and `PlayerControlView` sections. The ability to override drawables is similar to the ability to override layout files (it's just a different resource type), and applies to `PlayerView` and `PlayerControlView` as well. Documentation goes into the `gh-pages` branch [here](https://github.com/google/ExoPlayer/tree/gh-pages). There's a `README.md` that points to the resources needed to get started.

**Comment by nomisRev** [comment]
> We fixed some notification flicker issues in 2.10.0, but they only affected fairly old devices (KitKat), so that's probably not what you're seeing.

No, I am on `2.10.3`so I should have those changes. Seeing chronometer flickering on Android P, Pixel XL.

I'll make some time to contribute documentation for `PlayerNotificationManager` once I have a working version locally I can write about.

EDIT: It was indeed because the service was still in the foreground since API 21 this apparently overrides the `onGoing` flag. I was able to rely on `ongoing` in `fun onNotificationPosted(notificationId: Int, notification: Notification?, ongoing: Boolean)`. Would you also advice relying on that flag for `Noisy`, `WakeLock` and `WifiLock` compared to the technique described below?

OLD:

It seems to have started when I added `WakeLock` and `WifiLock` but I am seeing it getting unregistered right after the notification is updated.

```
NotificationKt$createNotificationManager:  isOnGoing: false
LocksKt$locks: Releasing WakeLock & WifiLock
NoisyReceiversKt$audioBecomingNoisyHandler: Unregistering noisy: registered=false
```

I am releasing/acquiring using `MediaControllerCompat.Callback()` and listening to changes in the metadata.

```
private fun onChanged(state: PlaybackStateCompat) = when (state.state) {
  PlaybackStateCompat.STATE_BUFFERING, PlaybackStateCompat.STATE_PLAYING -> acquire()
  PlaybackStateCompat.STATE_STOPPED, PlaybackStateCompat.STATE_PAUSED -> release()
  else -> {
  }
}
```

If I understood correctly to keep my audio player alive in the background I need to have acquired the locks during playing. So `onChanged` should be the same as `NotificationManager#getOngoing`? I think it'd be great if we could add an extension for this or some docs to clarify this because I am not 100% sure I am doing it correctly atm either. I couldn't find any `PlayerNotificationManager` that does this because even [UAMP](https://github.com/googlesamples/android-UniversalMusicPlayer) doesn't use `PlayerNoficationManager` or `WakeLock`/`WifiLock`.

Thoughts?

**Comment by ojw28** [comment]
I'm assuming based on the "EDIT" part of the post above that this issue is resolved. If it's not but the remaining problems are unrelated to providing the customizable icons, please file a separate issue.

Please do feel free to contribute documentation when you have a chance!

**Comment by nomisRev** [comment]
Hi, @ojw28 it was indeed resolved. I did not close it yet because I was hoping for a response in relation to my wakelock question.

I've had an amazing experience with ExoPlayer V2 so far, thanks to all the amazing contributors for this project! 👏 👏 👏 

I am still planning to contribute to the documentation since I've found ExoPlayer to have a lot of hidden gems, and amazing features that I feel deserve more light on `exoplayer.dev`.
